addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ContentSecurityPolicyScannerUnitTest.java

```diff
@@ -151,35 +202,118 @@ public void shouldAlertOnWildcardFrameAncestorsDirective() throws HttpMalformedH
                 alertsRaised.get(0).getDescription(),
                 equalTo(
                         "The following directives either allow wildcard sources (or ancestors), are not "
-                                + "defined, or are overly broadly defined: \nframe-ancestors"));
+                                + "defined, or are overly broadly defined: \nframe-ancestors"
+                                + "\n\nThe directive(s): frame-ancestors are among the directives that do not "
+                                + "fallback to default-src, missing/excluding them is the same as allowing anything."));
         assertThat(
                 alertsRaised.get(0).getEvidence(),
-                equalTo("frame-ancestors *; default-src 'self'"));
+                equalTo("frame-ancestors *; default-src 'self'; form-action 'none'"));
         assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_MEDIUM));
         assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
     }
 
     @Test
-    public void shouldNotAlertOnReasonableCsp() throws HttpMalformedHeaderException {
+    public void shouldNotAlertOnReasonableCsp() {
         // Given
-        HttpMessage msg = new HttpMessage();
-        msg.setRequestHeader("GET https://www.example.com/test/ HTTP/1.1");
-
-        msg.setResponseBody("<html></html>");
-        msg.setResponseHeader(
-                "HTTP/1.1 200 OK\r\n"
-                        + "Server: Apache-Coyote/1.1\r\n"
-                        + "Content-Security-Policy: default-src 'self'; script-src 'self' "
-                        + "storage.googleapis.com cdn.temasys.io cdn.tiny.cloud *.google-analytics.com; "
-                        + "style-src 'self' *.googleapis.com; font-src 'self' data: *.googleapis.com "
-                        + "fonts.gstatic.com; frame-ancestors 'none'; worker-src 'self'\r\n"
-                        + "Content-Type: text/html;charset=ISO-8859-1\r\n"
-                        + "Content-Length: "
-                        + msg.getResponseBody().length()
-                        + "\r\n");
+        HttpMessage msg = createHttpMessageWithReasonableCsp(HTTP_HEADER_CSP);
         // When
         scanHttpResponseReceive(msg);
         // Then
         assertThat(alertsRaised.size(), equalTo(0));
     }
+
+    @Test
+    public void shouldRaiseAlertOnCspIncludedViaXCSP() {
```

Comment:
Could use parameterized test for the header names. Name it `shouldRaiseAlertOnLegacyCspHeader`?